packages/fuels-core/src/types/wrappers/coin_type.rs

```diff
@@ -47,10 +44,10 @@ impl CoinType {
         }
     }
 
-    pub fn asset_id(&self) -> AssetId {
+    pub fn asset_id(&self) -> Option<AssetId> {
```

Comment:
In this case, can we remove `asset_id()` from `wrappers/input.rs`? The assumption here is that a `CoinType` is either a coin with some asset or a message with the base asset so we can unwrap and default to base downstream.
This doesn't hold up if we call `asset_id()` on `Input` since it could also be a `Contract`. Seems like a potential footgun.

Reply:
I wanted to do this and force users to match and then extract the asset_id. I needed some support :D I will do it now